doc/api/stream.md

```diff
@@ -1342,6 +1365,31 @@ class SourceWrapper extends Readable {
   }
 }
 ```
+*Note*: The `readable.push()` method is intended be called only by Readable
+Implemeters, and only from within the `readable.\_read()` method.
+
+#### Errors While Reading
+
+It is recommended that errors occurring during the processing of the
+`readable.\_read()` method are emitted using the `'error'` event rather than
+being thrown. Throwing an Error from within `readable.\_read()` can result in
+expected and inconsistent behavior depending on whether the stream is operating
+in flowing or paused mode. Using the `'error'` event ensures consistent and
+predictable handling of errors.
+
+```js
+const Readable = require('stream').Readable;
+
+const myReadable = new Readable({
+  read(size) {
+    try {
+      // do some work
+    } catch (err) {
+      process.nextTick(() => this.emit('error', err));
+    }
+  }
+});
+```
```

Comment:
Same thing here, I would not mention using try-catches anywhere in this document.